File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/ios/Classes/Model/IDOStepGoalModel.swift
//
//  IDOStepGoalModel.swift
//  protocol_channel
//
//  Created by hc on 2023/10/12.
//

import Foundation
/// Get daily step goal event number
@objcMembers
public class IDOStepGoalModel: NSObject, IDOBaseModel {
    /// Daily step goal
    public var step: Int
    /// Weekly step goal
    /// Valid when v2_support_set_step_data_type_03_03 is enabled
    public var stepWeek: Int

    enum CodingKeys: String, CodingKey {
        case step = "step"
        case stepWeek = "step_week"
    }

    public init(step: Int, stepWeek: Int) {
        self.step = step
        self.stepWeek = stepWeek
    }

    public func toJsonString() -> String? {
        if let jsonData = try? JSONEncoder().encode(self) {
            let jsonString = String(data: jsonData, encoding: .utf8)
            return jsonString
        }
        return nil
    }
}

File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/android/src/main/kotlin/com/idosmart/model/IDOStepGoalModel.kt
//
//  IDOStepGoalModel.kt
//  protocol_channel
//
//  Created by hedongyang on 2023/10/30.
//

package com.idosmart.model

import com.google.gson.GsonBuilder
import com.google.gson.annotations.SerializedName

///
/// Get daily step goal event number

open class IDOStepGoalModel(step: Int, stepWeek: Int) : IDOBaseModel {

    /// Daily step goal
    @SerializedName("step")
    var step: Int = step

    /// Weekly step goal
    /// Valid when v2_support_set_step_data_type_03_03 is enabled
    @SerializedName("step_week")
    var stepWeek: Int = stepWeek


    

    override fun toJsonString(): String {
        return GsonBuilder().create().toJson(this).toString()
    }
}
    